File: 2129-capitalize-the-title/2129-capitalize-the-title.cpp
class Solution {
public:
    string capitalizeTitle(string title) {
        
        for(int i=0; i<title.length();i++)
        {
            int start=i;
            
            while(i<title.length() && title[i]!=' ')
            {
                title[i]=tolower(title[i]);
                i++;
            }
            if(i-start>2)
            {
                title[start]=toupper(title[start]);
            }
        }
        return title;
    }
};

File: 2129-capitalize-the-title/2129-capitalize-the-title.java
class Solution {
    public String capitalizeTitle(String title) {
       
        char[] ch = title.toCharArray();
        
        for(int i=0; i<ch.length; i++)
        {
            int start=i;
            while(i<ch.length && ch[i]!=' ')
            {
                ch[i]=Character.toLowerCase(ch[i]);
                i++;
            }
            if(i-start>2)
            {
                ch[start]=Character.toUpperCase(ch[start]);
            }
        }
        
        return String.valueOf(ch);
    }
}